src/librustc_typeck/check/generator_interior.rs

```diff
@@ -54,12 +54,15 @@ impl<'a, 'gcx, 'tcx> InteriorVisitor<'a, 'gcx, 'tcx> {
             debug!("type in expr = {:?}, scope = {:?}, type = {:?}, count = {}, yield_span = {:?}",
                    expr, scope, ty, self.expr_count, yield_span);
 
-            if self.fcx.any_unresolved_type_vars(&ty) {
-                let mut err = struct_span_err!(self.fcx.tcx.sess, source_span, E0698,
-                    "type inside generator must be known in this context");
-                err.span_note(yield_span,
-                              "the type is part of the generator because of this `yield`");
-                err.emit();
+            if let Some((unresolved_type_span, unresolved_type)) =
+                self.fcx.get_unresolved_type_vars(&ty).pop()
+            {
+                // If unresolved type isn't a ty_var then unresolved_type_span is None
+                self.fcx.need_type_info_err_in_generator(
+                    unresolved_type_span.unwrap_or(yield_span), unresolved_type)
```

Comment:
Nit: can we put `unresolved_type` on the next line? I found this hard to read. I think rustfmt would do something like so:

```rust
self.fcx.need_type_info_err(
    xxx.unwrap_or(...),
    unresolved_type,
)
    .span_note(
        yield_span,
        "...",
    )
    .emit()
```